userland/userland_generic.ld

```diff
@@ -74,6 +74,7 @@ SECTIONS {
     .app_state :
     {
         KEEP (*(.app_state))
+        . = ALIGN(4); /* Make sure we're word-aligned here */
```

Comment:
Is the important thing here that subsequent sections _begin_ at aligned addresses or that this section itself is a whole number of words?

I'm _guessing_ it's the former? In which case I think it'd be better to move the align directive to the beginning.

It _may_ also be both? In which case I'd actually support ALIGN directives at both the beginning and end of the sections, that way each section can be independently correct (and possibly safely moved in the future without re-introducing problems).

Could you update the comment(s) to include why alignment is needed at each point? We try to be pretty pedantic with our linker scripts as they're complicated beasts.

Reply:
Agree, and I _think_ the issue is that the subsequent section _begin_ word-aligned. @fhars?

Reply:
Yes, the important thing in my case was that the sections start word aligned. I'm not that proficient with linker script subtleties, so I'd prefer if someone who actually understands the scripts and their interaction with elf2tbf to take over from here.

Reply:
I'll fix this one not and then I suggest we merge. Thanks for keeping this alive @fhars